src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -1,69 +1,131 @@
 package com.google.sps.meltingpot.data;
 
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.Query;
+import com.google.firebase.cloud.FirestoreClient;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
 public class FirestoreDB implements DBInterface {
-    public String getRecipeContent(String Id) {
-        return null;
-    }
-    public RecipeMetadata getRecipeMetadata(String Id) {
-        return null;
-    }
-    public String addRecipe(RecipeMetadata newRecipe, String newContent) {
-        return null;
-    }
-    public void deleteRecipe(String Id) {
-        return;
-    }
+  public String getRecipeContent(String Id) {
+    DocumentReference recipeRef = DBUtils.recipe(Id);
+    DocumentSnapshot recipe = DBUtils.blockOnFuture(recipeRef.get());
+    return recipe.getString(DBUtils.RECIPE_CONTENT_KEY);
+  }
 
-    public void editRecipeTitleContent(String editedTitle, String editedContent) {
-        return;
-    }
+  public RecipeMetadata getRecipeMetadata(String Id) {
+    DocumentReference recipeRef = DBUtils.recipeMetadata(Id);
+    DocumentSnapshot recipeMetadataSnapshot = DBUtils.blockOnFuture(recipeRef.get());
+    RecipeMetadata recipeMetadata = recipeMetadataSnapshot.toObject(RecipeMetadata.class);
```

Comment:
Could be condensed

Reply:
Yup, both that and getRecipeContent have been inlined now.